docs/src/mdx-components.tsx

```diff
@@ -8,14 +8,20 @@ import { PropsTable } from './components/reference/PropsTable';
 import { AttributesTable } from './components/reference/AttributesTable';
 import { CssVariablesTable } from './components/reference/CssVariablesTable';
 import { TableCode } from './components/TableCode';
+import { getChildrenText } from './getChildrenText';
 
 interface MDXComponents {
   [key: string]: React.FC<any> | MDXComponents;
 }
 
 export const mdxComponents: MDXComponents = {
   code: Code,
-  h1: (props) => <h1 className="mb-4 text-3xl font-bold" {...props} />,
+  h1: (props) => (
+    <React.Fragment>
+      <h1 className="mb-4 text-3xl font-bold" {...props} />
+      <title>{`${getChildrenText(props.children)} · Base UI`}</title>
```

Comment:
Hmm I think short titles are a much better experience for the end user.

<img width="441" alt="image" src="https://github.com/user-attachments/assets/d6ca0001-f555-42e4-a591-e7436e96cc16">

***

I see MUI docs do that, but I'm not a fan of how messy opening a bunch of tabs from mui.com feels

<img width="848" alt="image" src="https://github.com/user-attachments/assets/124ac3e6-eb4b-4414-be2d-24e3a16a95e7">

Reply:
I agree. It does look better in the browser's tabs.
@oliviertassinari, I remember you were suggesting richer titles in the past. Any opinions on this?

Reply:
We can put the SEO juice in the url

url: `/react-component-name`
pageTitle: `Component name • Base UI`

Reply:
We've agreed on the following for titles and URLs:

Dialog • Base UI
https://base-ui.com/react/dialog

Bonus, URLs are going to match import paths:
```jsx
// When we do kebab-casing
import { Dialog } from "@base-ui-components/react/dialog"

// Eventually
import { Dialog } from "@base-ui/react/dialog"
```

Reply:
> I remember you were suggesting richer titles in the past. Any opinions on this?

Right, so why do the pages of MUI follow those patterns? 

- On the SEO side, if we can trust Moz.com ([account](https://www.notion.so/mui-org/Accounts-945f0ccb9f7749ccb9e84b9eb9b13ec7#72a1e5b66fd84f20b7a9132e29ee5586)), it's a ranking factor. Because 

<img width="703" alt="SCR-20241127-neia" src="https://github.com/user-attachments/assets/e41a4a28-4fd6-4f98-9a04-0cbba5002f1a">

- On the UX side, we could change the title after the page is rendered?

Personally, I work with a small screen, my tabs are very quickly cut, I can't read them most of the time, I have learned to live without tab titles, so that's why I didn't bother so much. Or maybe it was because I always felt that onboarding is critical considering the duration people stay on a specific stack, and it's not annoying enough to leave so it helps win in the long term.